app/presenters/document/web_resource_presenter.rb

```diff
@@ -0,0 +1,175 @@
+        downloadable: downloadable?,
+        playable: playable?,
+        thumbnail: thumbnail,
+        play_url: play_url,
+        technical_metadata: technical_metadata,
+        download: {
+          url: download_url,
+          text: t('site.object.actions.download')
+        }
+      }.tap do |item|
+        if player.nil?
+          item[:is_unknown_type] = url
+        else
+          item[:"is_#{player}"] = true
+        end
+      end
+    end
+
+    def play_url
+      @play_url ||= begin
+        @record_presenter.iiif_manifesto || url
+      end
+    end
+
+    def url
+      @url ||= render_document_show_field_value('about')
+    end
+
+    def media_rights
+      @media_rights ||= begin
+        rights = render_document_show_field_value('webResourceEdmRights')
+        rights.blank? ? @record_presenter.media_rights : rights
+      end
+    end
+
+    def mime_type
+      @mime_type ||= render_document_show_field_value('ebucoreHasMimeType')
+    end
+
+    # Media type function normalises mime types
+    def media_type
+      @media_type ||= begin
+        if @record_presenter.iiif_manifesto
+          'iiif'
+        else
+          case (mime_type || '').downcase
+          when /^audio\//
+            'audio'
+          when /^image\//
+            'image'
+          when /^video\//
+            'video'
+          when /^text\//, /\/pdf$/
+            'text'
+          else
+            @record_presenter.render_document_show_field_value('type')
+          end
+        end
+      end
+    end
+
+    def edm_media_type
+      @edm_media_type ||= begin
+        record_type = @record_presenter.render_document_show_field_value('type')
+        if record_type == '3D' || media_type == 'iiif'
+          record_type
+        elsif media_type == 'audio'
+          'SOUND'
+        else
+          media_type.upcase
+        end
+      end
+    end
+
+    def use_media_proxy?
+      Rails.application.config.x.edm_is_shown_by_proxy &&
+        mime_type.present? &&
+        mime_type.match('image/').nil?
+    end
+
+    def download_url
+      @download_url ||= begin
+        if use_media_proxy?
+          Rails.application.config.x.edm_is_shown_by_proxy + @record.fetch('about', '/') + '?view=' + CGI.escape(url)
+        else
+          url
+        end
+      end
+    end
+
+    def technical_metadata
+      file_size = number_to_human_size(render_document_show_field_value('ebucoreFileByteSize')) || ''
+      {
+        mime_type: mime_type,
+        file_size: file_size.split(' ').first,
+        file_unit: file_size.split(' ').last,
+        codec: render_document_show_field_value('edmCodecName'),
+        width: render_document_show_field_value('ebucoreWidth'),
+        height: render_document_show_field_value('ebucoreHeight'),
+        size_unit: 'pixels',
+        runtime: render_document_show_field_value('ebucoreDuration'),
+        runtime_unit: 'seconds'
+      }
+    end
+
+    def playable?
```

Comment:
Cyclomatic complexity for playable? is too high. [9/6]<br>Assignment Branch Condition size for playable? is too high. [17.89/15]<br>Perceived complexity for playable? is too high. [10/7]